crates/forge/src/collecting.rs

```diff
@@ -0,0 +1,588 @@
+
+    Ok(TestCrate {
+        crate_root: tests_tmp_dir_path,
+        crate_name: "tests".to_string(),
+        crate_type: TestCrateType::Tests,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::scarb::ForgeConfig;
+    use assert_fs::fixture::PathCopy;
+    use assert_fs::TempDir;
+    use test_collector::ExpectedTestResult;
+
+    #[test]
+    fn runner_config_argument_precedence() {
+        let config_from_scarb = ForgeConfig {
+            exit_first: false,
+            fork: vec![],
+            fuzzer_runs: Some(1234),
+            fuzzer_seed: Some(1000),
+        };
+        let config = RunnerConfig::new(
+            Default::default(),
+            None,
+            false,
+            true,
+            Some(100),
+            Some(32),
+            &config_from_scarb,
+        );
+        assert_eq!(
+            config,
+            RunnerConfig {
+                workspace_root: Default::default(),
+                test_name_filter: None,
+                exact_match: false,
+                exit_first: true,
+                fork_targets: vec![],
+                fuzzer_runs: 100,
+                fuzzer_seed: 32,
+            }
+        );
+    }
+
+    #[test]
+    fn collecting_test_crates() {
+        let temp = TempDir::new().unwrap();
+        temp.copy_from("tests/data/simple_package", &["**/*.cairo", "**/*.toml"])
+            .unwrap();
+        let package_path = Utf8PathBuf::from_path_buf(temp.to_path_buf()).unwrap();
+
+        let test_crates = collect_test_crates(
+            &package_path,
+            "simple_package",
+            &package_path,
+            &TempDir::new().unwrap(),
+        )
+        .unwrap();
+
+        assert!(test_crates.contains(&TestCrate {
+            crate_root: package_path,
+            crate_name: "simple_package".to_string(),
+            crate_type: TestCrateType::Lib,
+        }));
+        assert!(test_crates
+            .iter()
+            .any(|tc| tc.crate_name == "tests" && tc.crate_type == TestCrateType::Tests));
+    }
+
+    #[test]
+    fn packing_tests() {
+        let temp = TempDir::new().unwrap();
+        temp.copy_from("tests/data/simple_package", &["**/*.cairo", "**/*.toml"])
+            .unwrap();
+        let package_path = Utf8PathBuf::from_path_buf(temp.to_path_buf()).unwrap();
+        let tests_path = package_path.join("tests");
+
+        let temp_dir = TempDir::new().unwrap();
+        let tests = pack_tests_into_one_file(&temp_dir, &tests_path).unwrap();
+        let virtual_lib_path = tests.crate_root.join("lib.cairo");
+        let virtual_lib_u8_content = std::fs::read(&virtual_lib_path).unwrap();
+        let virtual_lib_content = std::str::from_utf8(&virtual_lib_u8_content).unwrap();
+
+        assert!(virtual_lib_path.try_exists().unwrap());
+        assert!(virtual_lib_content.contains("mod contract;"));
+        assert!(virtual_lib_content.contains("mod ext_function_test;"));
```

Comment:
Test case: what if names are duplicates? It can't happen because files in dir always have unique names, right?

Reply:
So a test case for that doesn't make much sense